plugins/focus-tracker/index.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Allows users to see what screen readers would see.
+ */
+
+let Plugin = require("../base");
+
+let annotate = require("../shared/annotate")("tabIndex");
+
+const FOCUS_EVENT = "focusin";
+const BLUR_EVENT = "focusout";
+
+// this will let us get a shorter info panel that just
+// lets the user know we are tracking their focus
+const PANEL_OPTIONS = {
+  statusPanelView: true
+};
+
+// we're going to use focusin and focusout because they bubble
+const FOCUS_STATES = {
+  [FOCUS_EVENT]: "tota11y-outlined",
+  [BLUR_EVENT]: "tota11y-was-focused"
+};
+
+// we'll use this to make sure we don't apply the was-focused
+// indicator to our tota11y panels
+const IGNORE = "tota11y"
+
+// we're going to attempt to visualize the tab order of the page
+// based on the source order of the tabbable elements
+// this Array contains our tabbable element selectors
+const TABABLE_ELEMENTS = [
+    "a[href]",
+    "area[href]",
+    "iframe",
+    "input:not([disabled])",
+    "select:not([disabled])",
+    "textarea:not([disabled])",
+    "button:not([disabled])",
+    "[tabIndex]:not([tabIndex=\"-1\"])",
+    "[contenteditable]"
+];
+
+// convenient method to quickly remove any classes this
+// plugin applied
+// it is outside of the class because it doesn't really need
+// to access this and it lets us now worry about binding
+// our event handlers
+const removeFocusClasses = (element) => {
+  element.classList.remove(...Object.values(FOCUS_STATES));
+};
+
+require("./style.less");
+
+class FocusTracker extends Plugin {
+    constructor(...args) {
+      const options = Object.assign({}, args, { panel: PANEL_OPTIONS });
+
+      super(options);
+    }
+
+    getTitle() {
+        return "Focus Tracker";
+    }
+
+    getDescription() {
+        return "Keep track of what's been focused as you tab through the page";
```

Comment:
No work requested; just musing. I wonder if we can add i18n support to tota11y at some point to make it even more useful for folks.

Reply:
I think you definitely could. It might be nice to create an Issue to track moving all the strings like this into JSON (you could probably do one Strings file per plugin) so it's easier to translate and select which language you'd like to load the relevant strings for.